### DISCUSSION TITLE
Extended Migration Support

### DISCUSSION THREAD

**Discussion by slavakurilyak** [directus/directus#21961] [Feature Requests]
### Summary

Directus offers limited TypeScript support requiring manual schema implementation and migration, which can be time-consuming, especially for large projects with numerous collections

While Directus offers some [schema migration endpoints](https://directus.io/blog/announcing-schema-migrations), it would be great to see full migration control similar to [Payload CMS](https://payloadcms.com/docs/database/migrations)

### Motivation

Looking for automated ways to manage schema and migrations using Directus only.

### Detailed Design

N/A

### Requirements List

Commands similar to:

```
directus migrate
directus migrate:status
directus migrate:create
directus migrate:down
directus migrate:refresh
directus migrate:reset
directus migrate:fresh
```

### Drawbacks

TBD

### Alternatives

Lack of migration support means developers fallback to solutions like [Drizzle ORM](https://www.felixvemmer.com/en/blog/drizzle-orm-directus-typescript-sdk).

Drizzle ORM automates the schema generation process, saving developers a significant amount of time and effort. With commands like `drizzle-kit introspect:{dialect}`, developers can automatically generate TypeScript interfaces and Zod models from their database schema, eliminating the need for manual schema updates.

### Adoption Strategy

CLI, SDK, or API changes

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
